components/chainhook-cli/src/scan/stacks.rs

```diff
@@ -538,13 +556,19 @@ pub async fn consolidate_local_stacks_chainstate_using_csv(
 
     let _ = download_stacks_dataset_if_required(config, ctx).await;
 
-    let mut canonical_fork = get_canonical_fork_from_tsv(config, ctx).await?;
+    let stacks_db = open_readonly_stacks_db_conn(&config.expected_cache_path(), ctx).unwrap();
```

Comment:
Same here. It probably worth introducing a new helper doing the loop / retry.

Reply:
Done in commit 9cc2173b6c89b52b13fac9870dc1383ca09d8972